src/ctl/src/cmd_impl/await_tree/utils.rs

```diff
@@ -0,0 +1,253 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::{self, BufRead};
+use std::path::Path;
+
+use crate::cmd_impl::await_tree::tree::{SpanNodeView, SpanView, TreeView};
+
+/// Check `impl Display for StackTraceResponseOutput<'_>` for the format of the file.
+pub(crate) fn extract_actor_traces<P: AsRef<Path>>(
+    path: P,
+) -> anyhow::Result<HashMap<u32, String>> {
+    let file = File::open(path)?;
+    let reader = io::BufReader::new(file);
+
+    let mut actor_traces = HashMap::new();
+    let mut in_actor_traces = false;
+    let mut current_actor_id = None;
+    let mut current_trace = String::new();
+
+    for line in reader.lines() {
+        let line = line?;
+
+        // Detect the start of the Actor Traces section
+        if line == "--- Actor Traces ---" {
+            in_actor_traces = true;
+            continue;
+        }
+
+        // Stop parsing if a new section is encountered
+        if line.starts_with("---") && in_actor_traces {
+            if let Some(actor_id) = current_actor_id {
+                actor_traces.insert(actor_id, current_trace.trim().to_owned());
+            }
+            break;
+        }
+
+        // Parse Actor ID
+        if in_actor_traces && line.starts_with(">> Actor ") {
+            // Save the previous actor trace before processing the next one
+            if let Some(actor_id) = current_actor_id {
+                actor_traces.insert(actor_id, current_trace.trim().to_owned());
+            }
+            // Extract actor_id
+            if let Some(id_str) = line.strip_prefix(">> Actor ") {
+                if let Ok(actor_id) = id_str.trim().parse::<u32>() {
+                    current_actor_id = Some(actor_id);
+                    current_trace.clear(); // Clear trace for the next actor
+                }
+            }
+        } else if in_actor_traces {
+            // Accumulate trace content for the current actor
+            current_trace.push_str(&line);
+            current_trace.push('\n');
+        }
+    }
+
+    // Store the last actor's trace if any
+    if let Some(actor_id) = current_actor_id {
+        actor_traces.insert(actor_id, current_trace.trim().to_owned());
+    }
+
+    Ok(actor_traces)
+}
+
+/// The process of converting the tree to text is not lossless—information such as
+/// `node_id` will be lost. Consequently, this function can only restore information
+/// to the best extent possible. Fields like `current` and `node_id` cannot be
+/// recovered, but this loss does not affect our bottleneck detection. In the
+/// function, we will set all `node_id` values to 0 and `current` to 100.
+pub fn parse_tree_view_from_text(input: &str) -> anyhow::Result<TreeView> {
```

Comment:
TBH I didn't expect us to write a function to work atop of the human-readable format. But as it's proven feasible, I'm okay with it and we can keep it until json format is supported in all active RisingWave releases.

Agree on @xxchan that it would be better to maintain this in a separate repo (and thus a separate tool) for faster iteration.

Reply:
> TBH I didn't expect us to write a function to work atop of the human-readable format

Me neither. But AI can do it. 🤣